src/test/java/org/opensearch/knn/indices/ModelIndexTests.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.indices;
+
+import org.opensearch.ExceptionsHelper;
+import org.opensearch.ResourceAlreadyExistsException;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.admin.indices.create.CreateIndexResponse;
+import org.opensearch.action.delete.DeleteResponse;
+import org.opensearch.action.index.IndexRequest;
+import org.opensearch.action.index.IndexResponse;
+import org.opensearch.action.support.WriteRequest;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.index.engine.VersionConflictEngineException;
+import org.opensearch.knn.KNNSingleNodeTestCase;
+import org.opensearch.knn.index.util.KNNEngine;
+import org.opensearch.rest.RestStatus;
+
+import java.io.IOException;
+import java.util.Base64;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import static org.opensearch.knn.common.KNNConstants.MODEL_BLOB_PARAMETER;
+import static org.opensearch.knn.common.KNNConstants.MODEL_INDEX_NAME;
+
+public class ModelIndexTests extends KNNSingleNodeTestCase {
+
+    public void testCreate() throws IOException, InterruptedException {
+        int attempts = 50;
+        final CountDownLatch inProgressLatch = new CountDownLatch(attempts);
+
+        ActionListener<CreateIndexResponse> indexCreationListener = ActionListener.wrap(response -> {
+            assertTrue(response.isAcknowledged());
+            inProgressLatch.countDown();
+        }, exception -> {
+            if (!(ExceptionsHelper.unwrapCause(exception) instanceof ResourceAlreadyExistsException)) {
+                fail("Failed for reason other than ResourceAlreadyExistsException: " + exception);
+            }
+            inProgressLatch.countDown();
+        });
+
+        ModelIndex modelIndex = ModelIndex.getInstance();
+
+        for (int i = 0; i < attempts; i++) {
+            modelIndex.create(indexCreationListener);
+        }
+
+        assertTrue(inProgressLatch.await(30, TimeUnit.SECONDS));
+    }
+
+    public void testExists() {
+        ModelIndex modelIndex = ModelIndex.getInstance();
+        assertFalse(modelIndex.isCreated());
+        createIndex(MODEL_INDEX_NAME);
+        assertTrue(modelIndex.isCreated());
+    }
+
+    public void testPut() throws InterruptedException {
```

Comment:
nit: shall we split this into multiple test methods? Also, will the created model index is deleted at the end of the test?

Reply:
Yes, I can split into multiple tests

Yes the model index is deleted at the end of each test.